backend/lib/services/members.js

```diff
@@ -82,22 +130,64 @@ async function deleteServiceaccount (client, { namespace, name }) {
   }
 }
 
-async function setProjectMember (client, { namespace, name, roles: memberRoles }) {
+const removeAllOccurrencesOfMemberFromList = function (memberName, memberList) {
+  let member
+  do {
+    member = memberFromNotNormalizedProjectMembers(memberList, memberName)
+    if (member) {
+      _.remove(memberList, member)
+    }
+  } while (member)
+}
+
+// Adds or updates member, makes sure that a member occurs only once in the list and
+// migrates ServiceAccounts into kind ServiceAccount
+const updateMemberInNotNormalizedProjectMemberList = function (memberName, memberRoles, memberList) {
```

Comment:
I would assume the not normalized project member list in case it is not mentioned explicitly that is is normalized
-> rename to updateMember?

Reply:
you are right, of course. However, during development, I caught myself several times when I was confused about the members beeing normalized or not ;) So I added it just to be sure